network/hostport.go

```diff
@@ -236,3 +240,35 @@ func EnsureFirstHostPort(first HostPort, hps []HostPort) []HostPort {
 	result = append([]HostPort{first}, result...)
 	return result
 }
+
+// FastestHostPort tries to connect to all given hostPorts in parallel, waiting
+// up to the given timeout for each connection to be established, and using up
+// to GOMAXPROCS concurrent connection attempts. Returns the first HostPort
+// successfully connected to - the one with the lowest latency - or an error if
+// none of the given hostPorts can be reached.
+//
+// Timeout should be short, e.g. between 100ms and 1000ms.
```

Comment:
From the PR I guess nothing is using this but I wonder if the exemplary timeout value of 1s is too short on a heavily loaded system.